src/state.ts

```diff
@@ -0,0 +1,24 @@
+import {Event} from './event';
+
+export class State {
+	public readonly events;
+
+	constructor() {
+		this.events = {};
+	}
+
+	async save(event: Event): Promise<unknown> {
+		if (event.key && event.value) {
+			await this.events.push(event);
+		}
+		return this.events;
+	}
+
+	async retrieve(key): Promise<unknown> {
```

Comment:
Make retrieve a generic so that it’s called with a type:
ex: async function retrieve<T>(…)
we will want specific type helpers which pass a type in like retrieveString which returns retrieve<string>(…)